src/info.c

```diff
@@ -0,0 +1,179 @@
+        fun = DefaultInfoHandler;
+
+    CALL_3ARGS(fun, cls, lvl, args);
+}
+
+
+Obj InfoCheckLevel(Obj selectors, Obj level)
+{
+    if (STATE(ShowUsedInfoClassesActive)) {
+        CALL_2ARGS(ShowUsedInfoClassesHandler, selectors, level);
+    }
+    // Fast-path the most common failing case.
+    // The fast-path only deals with the case where all arguments are of the
+    // correct type, and were False is returned.
+    if (CALL_1ARGS(IsInfoClassListRep, selectors) == True) {
+#if defined(HPCGAP)
+        Obj index = ElmAList(selectors, INFODATA_CURRENTLEVEL);
+#else
+        Obj index = ELM_PLIST(selectors, INFODATA_CURRENTLEVEL);
+#endif
+        if (IS_INTOBJ(index) && IS_INTOBJ(level)) {
+            // < on INTOBJs compares the represented integers.
+            if (index < level) {
+                return False;
+            }
+        }
+    }
+    return CALL_2ARGS(InfoDecision, selectors, level);
+}
+
+/****************************************************************************
+**
+*F * * * * * * * * * * * * * initialize module * * * * * * * * * * * * * * *
+*/
+
+/****************************************************************************
+**
+*V  GVarFuncs . . . . . . . . . . . . . . . . . . list of functions to export
+*/
+static StructGVarFunc GVarFuncs[] = {
+    GVAR_FUNC(ShowUsedInfoClasses, 1, "choice"), { 0, 0, 0, 0, 0 }
+};
+
+
+/****************************************************************************
+**
+*F  InitKernel( <module> )  . . . . . . . . initialise kernel data structures
+*/
+static Int InitKernel(StructInitInfo * module)
+{
+    /* init filters and functions                                          */
+    InitHdlrFuncsFromTable(GVarFuncs);
+
+    /* The work of handling Info messages is delegated to the GAP level */
+    ImportFuncFromLibrary("InfoDecision", &InfoDecision);
+    ImportFuncFromLibrary("DefaultInfoHandler", &DefaultInfoHandler);
+    ImportFuncFromLibrary("IsInfoClassListRep", &IsInfoClassListRep);
+    ImportFuncFromLibrary("RESET_SHOW_USED_INFO_CLASSES",
+                          &ResetShowUsedInfoClassesHandler);
+    ImportFuncFromLibrary("SHOW_USED_INFO_CLASSES",
+                          &ShowUsedInfoClassesHandler);
+
+    /* return success                                                      */
+    return 0;
+}
+
+
+/****************************************************************************
+**
+*F  InitLibrary( <module> ) . . . . . . .  initialise library data structures
+*/
+static Int InitLibrary(StructInitInfo * module)
+{
+
+    InitGVarFuncsFromTable(GVarFuncs);
+
+    ExportAsConstantGVar(INFODATA_CURRENTLEVEL);
+    ExportAsConstantGVar(INFODATA_CLASSNAME);
+    ExportAsConstantGVar(INFODATA_HANDLER);
+    ExportAsConstantGVar(INFODATA_OUTPUT);
+    ExportAsConstantGVar(INFODATA_NUM);
+
+    /* return success                                                      */
+    return 0;
+}
+
+static Int InitModuleState(void)
+{
+    // return success
+    return 0;
+}
```

Comment:
Why add an `InitModuleState` function?

Reply:
Just cut+pasted some other file, assumed it was a required thing.